test/e2e/e2e_suite_test.go

```diff
@@ -47,10 +47,16 @@ var _ = BeforeSuite(func() {
 
 	if !config.IsSoakTest {
 		if !(config.IsUpgradeTest && helm.ReleaseExists()) {
```

Comment:
So, for PR jobs `IsUpgradeTest` will always be false because we run upgrade tests only with nightly. That means the e2e test will run twice with the "new" version both times. If we fix the comment above and make it conditional to execute the step, then this should be fixed.